api/shipper.proto

```diff
@@ -3,11 +3,8 @@ syntax = "proto3";
 option go_package = "github.com/elastic/elastic-agent-shipper/api";
 package elastic.agent.shipper.v1;
 
-
-
-import "google/protobuf/timestamp.proto";
-import "google/protobuf/struct.proto";
-import "google/rpc/status.proto";
+import "messages/publish.proto";
```

Comment:
What's the advantage of putting these in a separate package vs just separate files in the same package? Usually I would only use a separate package if I wanted to version them separately, otherwise it seems it just adds an extra import statement everywhere.

Reply:
The `struct` fork could perhaps live in a separate package because the license is different from the rest.

Reply:
On that note, we should probably have license headers on top of our own proto files.

Reply:
I think it's cleaner structured that message definitions are separate and `Struct` is a part of message definitions now we have it in our repo. I can't see any disadvantage of having them separately but I do see a lot of semantic value when you use it from Go code and extend messages in the future. I'm not a fan of having all the code in a single file.

Reply:
Use this license: https://github.com/elastic/elastic-agent-shipper/blob/18fb98d457d6dd32977f63572e16b8ba46866348/cmd/cmd.go#L1-L3

Our auto-licenser and lint check doesn't work for proto files it appears